diff --git a/src/ros_robot_localization_listener.cpp b/src/ros_robot_localization_listener.cpp
index 4038704..83b44d2 100644
--- a/src/ros_robot_localization_listener.cpp
+++ b/src/ros_robot_localization_listener.cpp
@@ -70,9 +70,9 @@ RosRobotLocalizationListener::RosRobotLocalizationListener(
   rclcpp::SubscriptionOptions options)
 : qos1_(1),
   qos10_(10),
-  odom_sub_(node, "odom/filtered", qos1_, options),
-  accel_sub_(node, "acceleration/filtered", qos1_, options),
-  sync_(10u, odom_sub_, accel_sub_),
+  odom_sub_(node, "odom/filtered", qos1_.get_rmw_qos_profile(), options),
+  accel_sub_(node, "acceleration/filtered", qos1_.get_rmw_qos_profile(), options),
+  sync_(odom_sub_, accel_sub_, 10u),
   node_clock_(node->get_node_clock_interface()->get_clock()),
   node_logger_(node->get_node_logging_interface()),
   base_frame_id_(""),
